src/ci/bin/test.inc.sh

```diff
@@ -215,6 +218,83 @@ cromwell::private::create_build_variables() {
+            CROMWELL_BUILD_URL="${CIRCLE_BUILD_URL}"
+            CROMWELL_BUILD_GIT_USER_EMAIL="builds@circleci.com"
+            CROMWELL_BUILD_GIT_USER_NAME="CircleCI"
+            CROMWELL_BUILD_HEARTBEAT_PATTERN="…"
+            CROMWELL_BUILD_GENERATE_COVERAGE=true
+            CROMWELL_BUILD_BRANCH="${CIRCLE_BRANCH:-${CIRCLE_TAG}}"
+            CROMWELL_BUILD_TAG="${CIRCLE_TAG:-}"
+
+            local circle_github_repository
+            circle_github_repository="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}"
+
+            if [[ "${circle_github_repository}" == "broadinstitute/cromwell" ]]; then
+                CROMWELL_BUILD_IS_SECURE=true
+            else
+                CROMWELL_BUILD_IS_SECURE=false
+            fi
+
+            if [[ "$BUILD_BRANCH_OR_PR" == "pr" ]]; then
+              CROMWELL_BUILD_EVENT="pull_request"
+            elif [[ "$BUILD_BRANCH_OR_PR" == "branch" ]]; then
+              CROMWELL_BUILD_EVENT="push"
+            else
+              echo "Allowed values for BUILD_BRANCH_OR_PR environment variable are 'branch' and 'pr', but got '${BUILD_BRANCH_OR_PR}'. Fallback to default 'branch' case."
+              CROMWELL_BUILD_EVENT="push"
+            fi
+
+            local circle_commit_message
+            local circle_force_tests
+            local circle_minimal_tests
+            if [[ -n "${CIRCLE_COMMIT_RANGE:+set}" ]]; then
+                # The commit message to analyze should be the last one in the commit range.
+                # This works for both pull_request and push builds, unlike using 'git log HEAD' which
+                # gives a merge commit message on pull requests.
+                circle_commit_message="$(git log --reverse "${CIRCLE_COMMIT_RANGE}" | tail -n1 2>/dev/null || true)"
+            fi
+
+            if [[ -z "${circle_commit_message:-}" ]]; then
+                circle_commit_message="$(git log --format=%B --max-count=1 HEAD 2>/dev/null || true)"
+            fi
+
+            if [[ "${circle_commit_message}" == *"[force ci]"* ]]; then
+                circle_force_tests=true
+                circle_minimal_tests=false
+            elif [[ "${circle_commit_message}" == *"[minimal ci]"* ]]; then
+                circle_force_tests=false
+                circle_minimal_tests=true
+            else
+                circle_force_tests=false
+                circle_minimal_tests=false
+            fi
+
+            echo "Building for commit message='${circle_commit_message}' with force=${circle_force_tests} and minimal=${circle_minimal_tests}"
+
+            # For solely documentation updates run only checkPublish. Otherwise always run sbt, even for 'push'.
+            # This allows quick sanity checks before starting PRs *and* publishing after merges into develop.
+            if [[ "${circle_force_tests}" == "true" ]]; then
+                CROMWELL_BUILD_RUN_TESTS=true
+            elif [[ "${CROMWELL_BUILD_ONLY_DOCS_CHANGED}" == "true" ]] && \
```

Comment:
If we meet our goal of making tests fast-ish and reliable we could consider dropping docs-only special handling to reduce complexity

Reply:
I agree, but I think it should be a separate follow-up ticket.
We could also remove `force ci` thing.

Reply:
Agreed.